app/javascript/controllers/opening_times_controller.js

```diff
@@ -0,0 +1,151 @@
+import { Controller } from "@hotwired/stimulus";
+
+const dayOrder = [
+	"Monday",
+	"Tuesday",
+	"Wednesday",
+	"Thursday",
+	"Friday",
+	"Saturday",
+	"Sunday",
+];
+
+// https://schema.org/OpeningHoursSpecification
+const openingHoursSpec = (day, open, close) => ({
+	"@type": "OpeningHoursSpecification",
+	dayOfWeek: `http://schema.org/${day}`,
+	opens: `${open}:00`,
+	closes: `${close}:00`,
+});
+
+const openingHoursObj = (openSpec) => ({
+	day: openSpec.dayOfWeek.split("/").filter((str) => str.includes("day"))[0],
+	open: openSpec.opens.slice(0, 5),
+	close: openSpec.closes.slice(0, 5),
+});
+
+const sortedOpeningHours = (openSpecArray) =>
+	[...openSpecArray]
+		.sort(
+			(a, b) =>
+				parseFloat(openingHoursObj(a).open.replace(":", ".")) -
```

Comment:
i would feel happier if all the `parseFloat(openingHoursObj(a).open.replace(":", "."))` was done once outside of the sort it you will have objects sent to the callback multiple times duplicating work

Reply:
I'm not sure this is possible using the lodash method? would it involve mutating the array, sorting it, then mutating it back?